openquake/calculators/event_based_risk.py

```diff
@@ -301,22 +281,23 @@ def pre_execute(self):
             self.agg_loss_table[l, r] = self.datastore.create_dset(
                 'agg_loss_table/rlz-%03d/%s' % (r, lt), self.elt_dt)
 
-    def execute(self):
-        """
-        Run the event_based_risk calculator and aggregate the results
-        """
         self.saved = collections.Counter()  # nbytes per HDF5 key
         self.ass_bytes = 0
         self.agg_bytes = 0
         self.gmfbytes = 0
         rlz_ids = getattr(self.oqparam, 'rlz_ids', ())
         if rlz_ids:
             self.rlzs_assoc = self.rlzs_assoc.extract(rlz_ids)
+
+        # NB: riskinputs is a generator
+        riskinputs = self.riskmodel.build_inputs_from_ruptures(
+            self.sitecol.complete, all_ruptures, oq.truncation_level,
+            correl_model, oq.minimum_intensity, eps, oq.concurrent_tasks or 1)
         return starmap(
             self.core_task.__func__,
             ((riskinput, self.riskmodel, self.rlzs_assoc,
               self.assetcol, self.monitor.new('task'))
-             for riskinput in self.riskinputs)).reduce(
+             for riskinput in riskinputs)).reduce(
```

Comment:
You might add a short comment also here, to highlight you are using the generator